### ISSUE TITLE
JsonUtils Fails to Convert to a Record Field with Enums

### ISSUE THREAD

**Issue by ThisaruGuruge** [ballerina-platform/ballerina-lang#35040]
**Description:**
When a record type has a field of type enum, the JsonUtils runtime API fails to convert the JSON value to a record type.

Consider the following Ballerina code:

```ballerina
enum Status {
    OPEN,
    CLOSE
}

type Shop record {
    string name;
    Status status;
};
```

Now, consider the following JSON:

```json
{
    "name": "My Shop",
    "status": "OPEN"
}
```

When we use the runtime `JsonUtils` class to convert this JSON to the record type as in the following code:

```java
BMap<BString, Object> recordValue = JsonUtils.converJSONToRecord(jsonMap, recordType);
```

The following error is thrown:

```
error while mapping 'status': incompatible types: expected 'thisaru/sample:Status', found 'string'
io.ballerina.runtime.internal.util.exceptions.BallerinaException: error while mapping 'status': incompatible types: expected 'thisaru/sample:Status', found 'string'
```

Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/2620

**Comment by Nadeeshan96** [comment]
@ThisaruGuruge Can you please check whether, you can use `cloneWithType` instead of `JsonUtils` convert methods? This is to avoid having issues in future, because the latest changes are there in the `cloneWithType` convert methods and `io.ballerina.runtime.internal.JsonUtils` is old and may be missing cases and necessary updates.

**Comment by ThisaruGuruge** [comment]
Is there an API named `cloneWithType`? Can you show me an example usage for that? In which class this API is?

I am referring to `io.ballerina.runtime.api.utils.JsonUtils` class. Not the internal one. We (stdlibs / people outside the lang repo) cannot use the internal APIs anyway I think.

**Comment by Nadeeshan96** [comment]
Since there is no runtime API to expose `cloneWithType` convert methods, created the task https://github.com/ballerina-platform/ballerina-lang/issues/35141 in order to do that.

**Comment by Nadeeshan96** [comment]
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/35103.

**Comment by ThisaruGuruge** [comment]
> Fixed with #35103.

Shouldn't this go with the patch? I see the fix is sent to the update branch, not the patch branch.

**Comment by Nadeeshan96** [comment]
> > Fixed with #35103.
> 
> Shouldn't this go with the patch? I see the fix is sent to the update branch, not the patch branch.

@warunalakshitha will merge the [runtime-swanlake-update-fixes](https://github.com/ballerina-platform/ballerina-lang/tree/runtime-swanlake-update-fixes) branch which contained the bug fixes to the [2201.0.x](https://github.com/ballerina-platform/ballerina-lang/tree/2201.0.x) branch after solving the merge conflicts. We are sending the update fixes to the [master](https://github.com/ballerina-platform/ballerina-lang) branch.
